Tools-Override/tests.targets

```diff
@@ -55,11 +61,71 @@
 
     <TestCommandLine Condition="'$(Performance)'!='true'">$(TestProgram) $(TestArguments) {XunitTraitOptions}</TestCommandLine>
   </PropertyGroup>
+  
+  <!-- This will generate a version of the dotnet cli that runs with the live
+       build of corefx using the other components gathered from the stable 
+       version consumed from the Tools directory
+       To debug any issues set COREHOST_TRACE=1 and rerun dotnet
+  -->
+  <UsingTask TaskName="GenerateDepsJson" AssemblyFile="$(BuildToolsTaskDir)CoreFx.Tools.dll"/>
+  <PropertyGroup>
+    <NewSharedFxVersion>9.9.9</NewSharedFxVersion>
+    <TestDotNetCliPath>$(ToolsDir)dotnetcli\shared\Microsoft.NETCore.App\$(NewSharedFxVersion)</TestDotNetCliPath>
+    <OldDotNetCliPath>$(ToolsDir)dotnetcli</OldDotNetCliPath>
+  </PropertyGroup>
+  <Target Name="ConstructTestingHost" Condition="!Exists('$(TestDotNetCliPath)')" >
+    <ItemGroup>
+      <LiveDotNetCliComponents Include="$(RuntimePath)\*.*" Exclude="$(RuntimePath)\xunit.console.netcore.exe" />
+      <DotNetCliSharedFrameworkSdkRuntimeConfig Include="$(OldDotNetCliPath)\sdk\**\dotnet.runtimeconfig.json" />
+      <DotNetCliHostPolicy Include="$(OldDotNetCliPath)\shared\**\hostpolicy.dll" />
```

Comment:
This only seems to work with a hostpolicy.dll in shared, the native code seems to have hard-coded a requirement to it. I could do something similar to the CoreClr packages and add an exclusion. Though I think we should be testing the one that came with the dotnet.exe it shipped with